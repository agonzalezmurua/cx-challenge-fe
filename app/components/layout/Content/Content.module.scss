.container {
  display: flex;
  justify-content: space-between;
  
  padding-top: 4rem;
}

.left {
  flex: 1;
  display: flex;
  justify-content: flex-end;
}

.middle {
  max-width: var(--max-width);
  flex-grow: 1;
}

.right{
  flex: 1
}