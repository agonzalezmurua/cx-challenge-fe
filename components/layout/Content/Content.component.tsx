import { Children } from "react";
import styles from "./Content.module.scss";

type ContentProps = {
  children: [JSX.Element, JSX.Element, JSX.Element];
};

export const Content = (props: ContentProps) => {
  const [left, middle, right] = Children.toArray(props.children);

  return (
    <section className={styles.container}>
      <section className={styles.left}>{left}</section>
      <section className={styles.middle}>{middle}</section>
      <section className={styles.right}>{right}</section>
    </section>
  );
};
