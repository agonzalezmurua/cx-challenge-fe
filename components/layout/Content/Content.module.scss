
@import "../../../variables.scss";

.container {
  --max-content-width: var(--max-width);
  --top-reserved-space: 4rem;

  display: flex;
  justify-content: space-between;
  
  padding-top: var(--top-reserved-space);

  @media only screen and (max-width: ($max-width + 420px)) {
    flex-direction: column;
    align-items: center;

    margin: auto;

    padding-top: 0;
    padding: 1rem;

    .left {
      width: 100%;

      max-width: var(--max-content-width);
      min-height: var(--top-reserved-space);

      padding-top: calc(1rem - var(--top-reserved-space));
      padding-bottom: 1rem;

      align-items: flex-end;
      justify-content: flex-start;

      z-index: 1;
    }
  }
}

.left {
  flex: 1;
  display: flex;
  justify-content: flex-end;
}

.middle {
  max-width: var(--max-content-width);
  flex-grow: 1;
  width: 100%;
}

.right{
  flex: 1
}