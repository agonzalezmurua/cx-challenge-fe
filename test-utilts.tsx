import { makeStore } from "@/redux/store";
import { queries, Queries } from "@testing-library/dom";
import { render, RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";

const store = makeStore({
  app: {
    parameters: {
      sorts: [
        {
          id: "irrelevant",
          name: "Menos relevante",
        },
      ],
      filters: [
        {
          id: "most_expensive",
          name: "Más caro",
          type: "price",
          values: [
            {
              id: "*-*",
              name: "infinito a infinito (infinito)",
              results: 10,
            },
          ],
        },
      ],
    },
    products: [
      {
        address: {
          city_name: "Caballito",
          state_name: "Capital Federal",
        },
        condition: "new",
        free_shipping: true,
        title: "Apple AirPods Con Estuche De Carga - Blanco",
        id: "MLA1363069275",
        installments: {
          amount: "6674.27",
          quantity: 6,
        },
        picture:
          "http://http2.mlstatic.com/D_923638-MLA54361048207_032023-I.jpg",
        price: {
          currency: "ARS",
          amount: "26994",
          decimals: 0,
        },
      },
    ],
    query: {
      price: "*-10000",
      search: "Apple Ipod",
      sort: { id: "relevance", name: "Más relevante" },
    },
  },
});

const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <Provider store={store}>{children}</Provider>;
};

const customRender = <
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container
>(
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
