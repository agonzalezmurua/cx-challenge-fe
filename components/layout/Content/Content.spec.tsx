import { render } from "@/test-utilts";
import { Content } from "./Content.component";

it("should be defined", () => {
  expect(Content).toBeDefined();
});

it("should render and match snapshot", () => {
  const result = render(
    <Content>
      <section>First</section>
      <section>Second</section>
      <section>Third</section>
    </Content>
  );

  expect(result.asFragment()).toMatchSnapshot();
  expect(result.getByText("First")).toBeInTheDocument();
  expect(result.getByText("Second")).toBeInTheDocument();
  expect(result.getByText("Third")).toBeInTheDocument();
});
