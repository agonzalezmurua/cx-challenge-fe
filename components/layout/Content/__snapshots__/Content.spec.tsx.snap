// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <section
      class="left"
    >
      <section>
        First
      </section>
    </section>
    <section
      class="middle"
    >
      <section>
        Second
      </section>
    </section>
    <section
      class="right"
    >
      <section>
        Third
      </section>
    </section>
  </section>
</DocumentFragment>
`;
